src/js/actions/layers.js

```diff
@@ -1372,9 +1372,16 @@ define(function (require, exports) {
                 var allLayerIDs = collection.pluck(results, "ID")
                     .reduce(function (allLayerIDs, layerIDs) {
                         return allLayerIDs.concat(layerIDs);
-                    }, []);
+                    }, []),
+                    allLayers = allLayerIDs.map(function (currentVal) {
+                        return { id: currentVal };
```

Comment:
It would be much better to just fetch the layer for the given ID. Not only would that be less surprising to the next person who reads this, but it would also save an object allocation per ID.